"Use of undeclared type or module `Needed`" in escaped_transform!

## Prerequisites

- Rust version : rustc 1.28.0-nightly (990d8aa74 2018-05-25)
- nom version : 4.0.0
- nom compilation features used: none

## Test case

Example test case:

```rust
#[macro_use]
extern crate nom;

named!(test<&str, String>,
  escaped_transform!(call!(alpha), '\\', tag!("n"))
);
```

Outputs:

```
error[E0433]: failed to resolve. Use of undeclared type or module `Needed`
   --> src/lib.rs:119:1
    |
119 | / named!(test<&str, String>,
120 | |   escaped_transform!(call!(alpha), '\\', tag!("n"))
121 | | );
    | |__^ Use of undeclared type or module `Needed`
```

--- a/src/bytes.rs
+++ b/src/bytes.rs
@@ -300,6 +300,7 @@ macro_rules! escaped_transform (
       use $crate::ExtendInto;
       use $crate::InputIter;
       use $crate::InputLength;
+      use $crate::Needed;
       use $crate::Slice;
       use $crate::need_more;
 

--- a/tests/issues.rs
+++ b/tests/issues.rs
@@ -272,3 +272,12 @@ fn issue_759() {
 named_args!(issue_771(count: usize)<Vec<u32>>,
   length_count!(value!(count), call!(nom::be_u32))
 );
+
+/// This test is in a separate module to check that all required symbols are imported in
+/// `escaped_transform!()`. Without the module, the `use`-es of the current module would
+/// mask the error ('"Use of undeclared type or module `Needed`" in escaped_transform!').
+mod issue_780 {
+  named!(issue_780<&str, String>,
+    escaped_transform!(call!(::nom::alpha), '\\', tag!("n"))
+  );
+}

